src/decoder/plugins/SidplayDecoderPlugin.cxx

```diff
@@ -437,51 +439,83 @@ sidplay_file_decode(DecoderClient &client, Path path_fs)
 	} while (cmd != DecoderCommand::STOP);
 }
 
+static AllocatedString<char>
+Windows1252ToUTF8(const char *s)
+{
+#ifdef HAVE_ICU_CONVERTER
+	try {
+		std::unique_ptr<IcuConverter>
+			converter(IcuConverter::Create("windows-1252"));
+
+		return converter->ToUTF8(s);
+	} catch (...) { }
+#endif
+
+	/*
+	 * Fallback to not transcoding windows-1252 to utf-8, that may result
+	 * in invalid utf-8 unless nonprintable characters are replaced.
+	 */
+	AllocatedString<char> t = AllocatedString<char>::Duplicate(s);
```

Comment:
For readability use `auto`

Reply:
Hmm... I can certainly change to `auto`. [ As a personal preference I consider `auto` _less_ readable though, because then one has to chase around to find what the type actually _is_. C++ isn't JavaScript or Python, after all. :smile: ]

Reply:
True, sometimes it's hard to tell; but sometimes, it doesn't matter as long as you get the idea what it can be used for (e.g. iterators), and in this case, it's too obvious (because the full type name is spelled again on the same line).